core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
Okay, so this is what happens. when we run `process_or_flush_pending_task` for 'emails' taskqueue task it will try to process it first, now if you see at the processing code in the else section then you see that `params=task.payload` is passed when processing function.

Now when we created a taskqueue task we did not add `payload`, we added `params` and both are different thing for a taskqueue. Now taskqueue's Task class has a property 'payload' which is equal to the payload we passed to it, but since we passed 'params' we need params and Task class has no property for getting params from an Task object.

So when taskqueue task (in this case) gets processed, it does not pass any params with it and thus the backend test (and probably functioning of whole services) fails. Because we need `user_id` parameter in reuqest, which is not there. Therefore when processing a taskqueue task we need to use a separate [code](https://github.com/prasanna08/oppia/blob/d41367bfab29bf04ad9cf38b7cf83ef612a023fd/core/controllers/feedback_test.py#L662-664).

Now since parameters will be different for different taskqueu(user_id in case of emails queue). So I think that we can not make a general function because from task object we cant get params of task.
Therefore what I did was just flush the task and not to process it and whenever required I processed it using code itself (above link).